File: collect_app/src/androidTest/java/org/odk/collect/android/support/pages/AddNewRepeatDialog.java
package org.odk.collect.android.support.pages;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.assertion.ViewAssertions.matches;
import static androidx.test.espresso.matcher.RootMatchers.isDialog;
import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
import static androidx.test.espresso.matcher.ViewMatchers.withText;

public class AddNewRepeatDialog extends Page<AddNewRepeatDialog> {

    private final String repeatName;

    public AddNewRepeatDialog(String repeatName) {
        this.repeatName = repeatName;
    }

    @Override
    public AddNewRepeatDialog assertOnPage() {
        String dialogMessage;
        if (repeatName.isBlank()) {
            dialogMessage = getTranslatedString(org.odk.collect.strings.R.string.add_another_question);
        } else {
            dialogMessage = getTranslatedString(org.odk.collect.strings.R.string.add_repeat_question, repeatName);
        }
        onView(withText(dialogMessage))
                .inRoot(isDialog())
                .check(matches(isDisplayed()));
        return this;
    }

    public <D extends Page<D>> D clickOnAdd(D destination) {
        return clickOnTextInDialog(org.odk.collect.strings.R.string.add_repeat, destination);
    }

    public <D extends Page<D>> D clickOnDoNotAdd(D destination) {
        return clickOnTextInDialog(org.odk.collect.strings.R.string.cancel, destination);
    }

}

File: collect_app/src/androidTest/java/org/odk/collect/android/support/pages/AddNewRepeatDialog.kt
package org.odk.collect.android.support.pages

import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.RootMatchers.isDialog
import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
import androidx.test.espresso.matcher.ViewMatchers.withText
import org.odk.collect.strings.R

class AddNewRepeatDialog(private val repeatName: String?) : Page<AddNewRepeatDialog>() {
    override fun assertOnPage(): AddNewRepeatDialog {
        val dialogMessage = if (repeatName.isNullOrBlank()) {
            getTranslatedString(R.string.add_another_question)
        } else {
            getTranslatedString(R.string.add_repeat_question, repeatName)
        }
        onView(withText(dialogMessage))
            .inRoot(isDialog())
            .check(matches(isDisplayed()))
        return this
    }

    fun <D : Page<D>> clickOnAdd(destination: D): D {
        return clickOnTextInDialog(R.string.add_repeat, destination)
    }

    fun <D : Page<D>> clickOnDoNotAdd(destination: D): D {
        return clickOnTextInDialog(R.string.cancel, destination)
    }
}
